File: example/src/main/java/com/github/vivchar/example/widgets/NestedAdapter.java
package com.github.vivchar.example.widgets;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.util.Log;
import android.view.ViewGroup;

import com.github.vivchar.rendererrecyclerviewadapter.RendererRecyclerViewAdapter;
import com.github.vivchar.rendererrecyclerviewadapter.ViewHolder;

import java.util.List;

/**
 * Created by Vivchar Vitaly on 21.10.17.
 */

public class NestedAdapter extends RendererRecyclerViewAdapter {

	private static final String TAG = NestedAdapter.class.getSimpleName();

	public NestedAdapter() {
		super();
	}

	@NonNull
    @Override
	public ViewHolder onCreateViewHolder(@NonNull final ViewGroup parent, final int typeIndex) {
		final ViewHolder viewHolder = super.onCreateViewHolder(parent, typeIndex);
		Log.d(TAG, "onCreateViewHolder: " + viewHolder.getClass().getSimpleName());
		return viewHolder;
	}

	@Override
	public void onBindViewHolder(@NonNull final ViewHolder holder, final int position, @Nullable final List payloads) {
		Log.d(TAG, "onBindViewHolder: " + holder.getClass().getSimpleName());
		super.onBindViewHolder(holder, position, payloads);
	}

	@Override
	public void onViewRecycled(final ViewHolder holder) {
		Log.d(TAG, "onViewRecycled: " + holder.getClass().getSimpleName());
		super.onViewRecycled(holder);
	}
}

File: example/src/main/java/com/github/vivchar/example/widgets/NestedAdapter.kt
package com.github.vivchar.example.widgets

import android.util.Log
import android.view.ViewGroup
import com.github.vivchar.rendererrecyclerviewadapter.ViewHolder

/**
 * Created by Vivchar Vitaly on 21.10.17.
 */
class NestedAdapter : MyAdapter() {

	override fun onCreateViewHolder(parent: ViewGroup, typeIndex: Int): ViewHolder<*> {
		return super.onCreateViewHolder(parent, typeIndex).apply {
			Log.d(TAG, "onCreateViewHolder: " + javaClass.simpleName)
		}
	}

	override fun onBindViewHolder(holder: ViewHolder<*>, position: Int) {
		super.onBindViewHolder(holder, position)
	}

	override fun onBindViewHolder(holder: ViewHolder<*>, position: Int, payloads: MutableList<Any?>) {
		Log.d(TAG, "onBindViewHolder: " + holder.javaClass.simpleName)
		super.onBindViewHolder(holder, position, payloads)
	}

	override fun onViewRecycled(holder: ViewHolder<*>) {
		Log.d(TAG, "onViewRecycled: " + holder.javaClass.simpleName)
		super.onViewRecycled(holder)
	}

	companion object {
		private val TAG = NestedAdapter::class.java.simpleName
	}
}